astropy/table/index.py

```diff
@@ -853,7 +864,7 @@ def _get_rows(self, item):
             # item should be a list or ndarray of values
             rows = []
             for key in item:
-                p = index.find((key,))
+                p = index.find(key if isinstance(key, tuple) else (key,))
```

Comment:
@taldcroft this part of the fix is yours. Let me know if I should explicitly add you as a co-author of the commit.